ccron/src/ticks_generator.cpp

```diff
@@ -93,9 +108,14 @@ void TicksGenerator::run() {
 void TicksGenerator::sendClientRequestMsgTick(std::uint32_t component_id) {
   auto payload = std::vector<std::uint8_t>{};
   serialize(payload, ClientReqMsgTickPayload{component_id});
-  const auto seq_num = bft_client_->sendRequest(
-      bftEngine::TICK_FLAG, payload.size(), reinterpret_cast<const char*>(payload.data()), kTickCid);
-  component_pending_req_seq_nums_[component_id] = seq_num;
+  const auto seq_num =
+      bft_client_->sendRequest(bftEngine::TICK_FLAG,
+                               payload.size(),
+                               reinterpret_cast<const char*>(payload.data()),
+                               kTickCid,
+                               [p = shared_from_this(), component_id]() { p->onTickPoppedFromExtQueue(component_id); });
+  pending_tick_requests_[component_id] = seq_num;
```

Comment:
I probably miss something, but I would have expected also a call to unset ```pending_tick_requests_[component_id] = seq_num``` on handling, no?

Reply:
The idea to keep it is that we map from component ID -> BFT seq number. In order to check if the request is still pending in the clients manager, we need the BFT sequence number if the request is not in the external queue:

```
    // If the tick for the given component ID is not in the external queue, check if it is pending. If yes, don't send.
    if (!pending_req_->isPending(bft_client_->getClientId(), req_it->second)) {
      sendClientRequestMsgTick(internal_tick.component_id);
    }
```

Does that make sense?

Reply:
I have added a diagram in the `onInternalTick()` method that explains it better. Thanks to @HristoStaykov 's comment that suggested adding it.

Reply:
I see, and after consensus is finished, does it stay there? can it cause an issue?

Reply:
It shouldn't, because we assume the internal BFT client doesn't generate duplicate BFT sequence numbers and, therefore, an already committed request would never be pending again.